api-put-object-streaming.go

```diff
@@ -270,14 +270,8 @@ func (c Client) putObjectMultipartStreamNoChecksum(bucketName, objectName string
 
 		var objPart ObjectPart
 		objPart, err = c.uploadPart(bucketName, objectName, uploadID,
-			io.LimitReader(hookReader, partSize), partNumber,
-			nil, nil, partSize, metadata)
-		// For unknown size, Read EOF we break away.
-		// We do not have to upload till totalPartsCount.
-		if err == io.EOF && size < 0 {
-			break
-		}
```

Comment:
Streaming we never support size == -1,  for size == -1 we fall back to regular multipart upload.